fuse/nodefs/fsconnector.go

```diff
@@ -126,9 +135,23 @@ func (c *FileSystemConnector) forgetUpdate(nodeID uint64, forgetCount int) {
 
 	if forgotten, handled := c.inodeMap.Forget(nodeID, forgetCount); forgotten {
 		node := (*Inode)(unsafe.Pointer(handled))
+		if node.IsDir() && len(node.Children()) > 0 {
+			// This directory node has children that have not yet been forgotten.
+			// We don't want to remove it from the tree (i.e. remove it from its parent)
+			// because the children may have parent pointers to it.
+			// A later LOOKUP will return this node with a new NodeID.
+			return
+		}
 		node.mount.treeLock.Lock()
-		c.recursiveConsiderDropInode(node)
+		// Remove Inode from all parents
```

Comment:
From what I know it already looks like 2.

Op wo 23 mrt. 2016 10:16 schreef rfjakob <notifications@github.com>:

> In fuse/nodefs/fsconnector.go
> <https://github.com/hanwen/go-fuse/pull/103#discussion_r57128289>:
>
> >  		node.mount.treeLock.Lock()
> > -		c.recursiveConsiderDropInode(node)
> > +		// Remove Inode from all parents
>
> So I see three options:
> 1) Have the full DAG in nodefs (= add parent pointers).
> 2) Have the "down" direction (children) of the DAG in nodefs, the "up"
> direction (parents) in pathfs. Looks a bit strange but should work.
> 3) Have the full DAG in pathfs (= add children and parent pointers)
>
> From what I understand you favor (2) ?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/hanwen/go-fuse/pull/103/files/004bb8a47f407fea15ba2caedcb216371ca806f4#r57128289>
>